src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -1273,7 +1287,10 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
 
     const settings = stream.getTracks()[0].getSettings();
     // If a device does not specify deviceId, we have to assume the stream is not reusable.
-    return settings.deviceId === this.getIntrinsicDeviceIdStr(device);
+    return (
+      settings.deviceId &&
+      settings.deviceId === DefaultDeviceController.getIntrinsicDeviceId(device)
+    );
```

Comment:
Nice. Was there a bug hiding here?

Reply:
Tbh, I am not sure how to make deviceId from mediaTrackSettings to be null, undefined, or empty string from the demo app but yes in theory it can be any of those value so it is better to guard against it.